bip-0141.mediawiki

```diff
@@ -127,7 +127,7 @@ Sigops per block is currently limited to 20,000. We change this restriction as f
 Sigops in the current pubkey script, signature script, and P2SH check script are counted at 4 times their previous value.
 The sigop limit is likewise quadrupled to ≤ 80,000.
 
-Each P2WPKH input is counted as 1 sigop. In addition, opcodes within a P2WSH <code>witnessScript</code> are counted identically as previously within the P2SH <code>redeemScript</code>. That is, CHECKSIG is counted as only 1 sigop, and CHECKMULTISIG is counted as 1 to 20 sigops according to the arguments. This rule applies to both native witness program and P2SH witness program.
+Each P2WPKH input is counted as 1 sigop. In addition, opcodes within a P2WSH <code>witnessScript</code> are counted identically as previously within the P2SH <code>redeemScript</code>. That is, CHECKSIG is counted as only 1 sigop. For 1 to 16 total public keys CHECKMULTISIG is counted as 1 to 16 sigops respectively, and for 17 to 20 total public keys it is counted as 20 sigops. This rule applies to both native witness program and P2SH witness program.
```

Comment:
I'd be more specific and say this rule only applies to OP_CHECKMULTISIG and OP_CHECKMULTISIGVERIFY preceded immediately by OP_1 through OP_16. That implicitly excludes 17-20, but it also makes it clear that if you'd use a direct push for 1-16 (or OP_PUSHDATA, or some other construction) it won't work.

Reply:
That's much clea[n/r]er, updated.